### PR TITLE
Update swc_core to `v0.75.23`

### LINKED ISSUE #46989: [NEXT-939] SIGSEGV on `next build` with 13.2.4
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: 7.26.0
    Relevant packages:
      next: 13.2.4
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected?

`next build` CLI

### To Reproduce

I'll dig into it, maybe I can reproduce it easily. However I'd expect `next build` not to segfault on my existing codebase which works on 13.2.3 without changes other than upgrading next.

### Describe the Bug

After upgrading next to 13.2.4, when running `next build`, I'm running into a segfault.

As you can see, I'm using the app dir (only). Tell me if you need more info.

```
$ next build --experimental-app-only
info  - Loaded env from /next/apps/web/.env.local
info  - Loaded env from /next/apps/web/.env.production
warn  - You have enabled experimental features (appDir, swcPlugins, serverComponentsExternalPackages) in next.config.mjs.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

(node:83878) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
error Command failed with signal "SIGSEGV".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /user/.nvm/versions/node/v18.13.0/bin/node
Arguments: /user/.cache/node/corepack/yarn/1.22.19/lib/cli.js build
Directory: /next
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
```

### Expected Behavior

No Segfault when running `yarn build`

### How are you deploying your application? (if relevant)

On Vercel, where it's also segfaulting. (Also in my Docker build)

<sub>[NEXT-939](https://linear.app/vercel/issue/NEXT-939/sigsegv-on-next-build-with-1324)</sub>

### PR DESCRIPTION
**Pull request by padmaia**
Depends on https://github.com/vercel/turbo/pull/4470

---

 - Closes https://github.com/vercel/next.js/issues/46989
 - Fixes WEB-879
 - Fixes WEB-813

### CODE REVIEW

**Review by jridgewell — DISMISSED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 981a28e51db3b310c3ed4c0362e63c655e119e2c

`pnpm testheadless test/integration/auto-export-error-bail/test/index.test.js` 
- Auto Export _error bail > server mode > should not opt-out of auto static optimization from invalid _error

<details>
<summary>Expand output</summary>

  ● Auto Export _error bail › server mode › should not opt-out of auto static optimization from invalid _error

    expect(received).toBe(expected) // Object.is equality

    Expected: 0
    Received: 1

      19 |     const combinedOutput = output.stderr + output.stdout
      20 |
    > 21 |     expect(output.code).toBe(0)
         |                         ^
      22 |     expect(combinedOutput).not.toContain(
      23 |       'You have opted-out of Automatic Static Optimization due to'
      24 |     )

      at Object.<anonymous> (integration/auto-export-error-bail/test/index.test.js:21:25)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/server/post-process.ts [hunk 1]
```diff
@@ -199,7 +199,7 @@ async function postProcessHTML(
       : null,
     process.env.NEXT_RUNTIME !== 'edge' && renderOpts.optimizeFonts
       ? async (html: string) => {
-          const getFontDefinition = (url: string): string => {
+          const getFontDefinition = (url: string) => {
```
Comment (kdy1) [inline packages/next/src/server/post-process.ts:202]: I filed [WEB-875](https://linear.app/vercel/issue/WEB-875)

Comment (kdy1) [reply packages/next/src/server/post-process.ts:202]: Will be fixed by https://github.com/swc-project/swc/pull/7232

**Comment by Himself65** [comment]
any updates?

**Comment by kdy1** [comment]
# Further investigation:

 - `processBinaryChunk` is from react.
 - Patching `ssr` to false from `next/dynamic` does not work.
 - Execution order is a bit strange

```
LoadableOptions {
  loading: [Function: loading],
  ssr: false,
  modules: [ 'legacy/no-ssr.js -> ../../ui/pure-client' ]
}
isServerSide true
Loading:  [Function: loading]
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at U (/private/var/folders/pw/7l7d6f6s6ngb9z17pc912mtc0000gn/T/next-install-5bc17e2a036d2273d8f55cf46c46aea781da64281b1000bf9cb7582d1923ca4b/node_modules/.pnpm/file+..+next-repo-1c149e8656005f23f855f684eb57e24c_ea2336ebcj5zuuo46igxqabumu/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:22:196)
    at c (/private/var/folders/pw/7l7d6f6s6ngb9z17pc912mtc0000gn/T/next-install-5bc17e2a036d2273d8f55cf46c46aea781da64281b1000bf9cb7582d1923ca4b/node_modules/.pnpm/file+..+next-repo-1c149e8656005f23f855f684eb57e24c_ea2336ebcj5zuuo46igxqabumu/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:25:307)

Error occurred prerendering page "/dynamic". Read more: https://nextjs.org/docs/messages/prerender-error
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at U (/private/var/folders/pw/7l7d6f6s6ngb9z17pc912mtc0000gn/T/next-install-5bc17e2a036d2273d8f55cf46c46aea781da64281b1000bf9cb7582d1923ca4b/node_modules/.pnpm/file+..+next-repo-1c149e8656005f23f855f684eb57e24c_ea2336ebcj5zuuo46igxqabumu/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:22:196)
    at c (/private/var/folders/pw/7l7d6f6s6ngb9z17pc912mtc0000gn/T/next-install-5bc17e2a036d2273d8f55cf46c46aea781da64281b1000bf9cb7582d1923ca4b/node_modules/.pnpm/file+..+next-repo-1c149e8656005f23f855f684eb57e24c_ea2336ebcj5zuuo46igxqabumu/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:25:307)
info  - Generating static pages (0/4)
el: {
  '$$typeof': Symbol(react.element),
  type: [Function: loading],
  key: null,
  ref: null,
  props: { error: null, isLoading: true, pastDelay: false, timedOut: false },
  _owner: null
}
Render loading:  { error: null, isLoading: true, pastDelay: false }
info  - Generating static pages (1/4)
info  - Generating static pages (2/4)

```

**Comment by kdy1** [comment]
# Further investigation:

`__next_f` of `((self as any).__next_f = (self as any).__next_f || [])`  has wrong data. I'll check code responsible for `__next_f`

**Comment by kdy1** [comment]
# Investigation

Some modules in `next.js` are removed

## ssrModuleMapping

image

## clientModules

(next.js part)

image
image
image
image

(e2e test app part)

image

File: test/production/escheck-output/index.test.ts [hunk 2]
```diff
@@ -23,29 +22,4 @@ describe('ES Check .next output', () => {
       'info: ES-Check: there were no ES version matching errors!  🎉'
     )
   })
-
-  it('should emit ES5 with legacyBrowsers: true', async () => {
```
Comment (kdy1) [inline test/production/escheck-output/index.test.ts:27]: This test is fundamentally broken.

x-ref: https://vercel.slack.com/archives/C04DUD7EB1B/p1681534087348619

File: packages/next/src/server/next.ts [hunk 3]
```diff
@@ -252,7 +252,7 @@ function createServer(options: NextServerOptions): NextServer {
 
 // Support commonjs `require('next')`
 module.exports = createServer
-exports = module.exports
+// exports = module.exports
```
Comment (timneutkens) [inline packages/next/src/server/next.ts:255]: Seems this is a bug?

Comment (kdy1) [reply packages/next/src/server/next.ts:255]: The problem is that when it's executed, `exports` is overridden.

```js
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    NextServer: function() {
        return NextServer;
    },
    // Support `import next from 'next'`
    default: function() {
        return _default;
    }
});
```

This is the generated code for exports, and those are removed by

```js
// Support commonjs `require('next')`
module.exports = createServer;
exports = module.exports;
const _default = createServer;

```

at the end of the file.

Comment (kdy1) [reply packages/next/src/server/next.ts:255]: But it's not a bug. ESM mixed with common js does not have the _correct_ behavior.

**Review by sokra — DISMISSED** [review]

**Review by kdy1 — APPROVED** [review]
https://github.com/vercel/turbo/pull/4470 is approved/merged along with this PR

**Comment by Kinbaum** [comment]
Hello, was hoping to get some assistance with an issue that I believe this PR introduced in `13.3.1-canary.12`.
Up until this point, I have been using cypress for component testing.

When a component contained a hook like `usePathname` or `useRouter`, I was able to successfully mock those by doing the following:

```js
import * as NextNavigation from 'next/navigation';

describe('test...', () => {
    it('should...', () => {
        const pathname = cy.stub().as('usePathname').returns('/en/public');
        cy.stub(NextNavigation, 'usePathname').returns(pathname());
    })
})
```

This approach worked for me up until this canary version was release. Now it throws the following error:

```
(uncaught exception) Error: invariant expected app router to be mounted
```

Does anyone have a suggestion on how to resolve this?

**Comment by Kinbaum** [comment]
Update: for those who come across this as well, I managed to solve by creating my own mount function specifically for components that depend on these hooks. Here's an abbreviated implementation:

```ts
import { MountOptions, MountReturn, mount } from 'cypress/react18';
import { AppRouterContext } from 'next/dist/shared/lib/app-router-context';
import { PathnameContext } from 'next/dist/shared/lib/hooks-client-context';

declare global {
    namespace Cypress {
        interface Chainable {
            mount: typeof mount;
            nextMount: (jsx: React.ReactNode, options?: Partial<MountOptions & {
                pathname?: string;
            }>) => Chainable<MountReturn>;
        }
    }
}

Cypress.Commands.add('mount', mount);

Cypress.Commands.add('nextMount', (component, options) => {
    const router = {
        back: cy.stub().as('router:back'),
        forward: cy.stub().as('router:forward'),
        prefetch: cy.stub().as('router:prefetch'),
        push: cy.stub().as('router:push'),
        reload: cy.stub().as('router:reload'),
        refresh: cy.stub().as('router:refresh'),
        replace: cy.stub().as('router:replace')
    };

    const pathname = options?.pathname ?? '/';

    return mount(
        <AppRouterContext.Provider value={ router }>
            <PathnameContext.Provider value={ pathname }>
                { component }
            </PathnameContext.Provider>
        </AppRouterContext.Provider>
    );
});
```

**Pull request merged**